packages/qwik-city/src/middleware/request-handler/request-event.ts

```diff
@@ -220,6 +229,21 @@ export function createRequestEvent(
       return new RedirectMessage();
     },
 
+    rewrite: (_rewriteUrl: string | URL) => {
+      check();
+
+      const rewriteUrl = typeof _rewriteUrl === 'string' ? _rewriteUrl : _rewriteUrl.toString();
+      const fixedURL = rewriteUrl.replace(/([^:])\/{2,}/g, '$1/');
+      if (rewriteUrl !== fixedURL) {
+        console.warn(`Rewrite URL ${rewriteUrl} is invalid, fixing to ${fixedURL}`);
+      }
+      url.pathname = fixedURL;
```

Comment:
// TODO - handle rewrite to urls with search params
      // This is not accurate, because the url might have search params

Reply:
Hmm. Ok, I can see rewriting to a string be like a shortcut that passes on search params, and then passing a URL object should not do anything

Reply:
I made the API as flexible as possible. Let me know if you think it should be different.